crates/p2p_proto/src/lib.rs

```diff
@@ -59,6 +65,12 @@ impl<M, T: ToProtobuf<M>> ToProtobuf<Vec<M>> for Vec<T> {
     }
 }
 
+impl<M, T: ToProtobuf<M>> ToProtobuf<Option<M>> for Option<T> {
```

Comment:
Did you try `impl TryFromProtobuf` for `Option<T>`? It could allow non-custom deserializer AFAIU.

Reply:
As per prev comment:
```
error[E0119]: conflicting implementations of trait `TryFromProtobuf<Option<_>>`
```

Reply:
:arrow_up: fixed with a field-annotation as per prev thread :arrow_up: